### ISSUE TITLE
Missing source code comments

### ISSUE THREAD

**Issue by chamika** [systers/PCSA-Android#94]
As per the coding section in the systers development process page, the source code needs to have proper comments such as "what does function do, what are the parameters, what does it return". But the current source code doesn't have that kind of commenting in some places.

Shall I add the comments in necessary places?

**Comment by sandarumk** [comment]
Hi Chamika, Can you define a comment template for methods and functions which adapts the common Android standards and the systers development standards?

**Comment by chamika** [comment]
The javadoc standard method comment is like this.

```
/**
 * Returns an Image object that can then be painted on the screen. 
 * The url argument must specify an absolute {@link URL}. The name
 * argument is a specifier that is relative to the url argument. 
 * <p>
 * This method always returns immediately, whether or not the 
 * image exists. When this applet attempts to draw the image on
 * the screen, the data will be loaded. The graphics primitives 
 * that draw the image will incrementally paint on the screen. 
 *
 * @param  url  an absolute URL giving the base location of the image
 * @param  name the location of the image, relative to the url argument
 * @return      the image at the specified URL
 * @see         Image
 */
 public Image getImage(URL url, String name) {
        try {
            return getImage(new URL(url, name));
        } catch (MalformedURLException e) {
            return null;
        }
 }
```

We can use the same format since it is the standard way

**Comment by sandarumk** [comment]
First go through the Reporting type process associated pages and submit a PR with the completed comments. Let's see how that works out and then if it looks ok, you can proceed with the whole application.

**Comment by chamika** [comment]
Before I implement the given comment template for methods, I would like to mention the lack of class comments. What about the licencing? Usually, a class contains the licensing information. I hope to follow the following template for the class comments

```
/*
 * Legal/copyright ownership information
 *
 * Proper comment on the class like what it does and 
 * if there are design level description
 * 
 * @author <Authorname>
 * @since 2016-03-29 
 */
```
Shall I include class comments for this as well?

Another developer (@rohands) suggested that this can be a great issue for beginners. I agree with him. So once I finish the Reporting Steps, you can tag this as a free issue and consider as a beginner task and ask them to follow the Reporting steps and given method and class templates.

**Comment by sandarumk** [comment]
Forget the licensing for now. You can proceed with the class comments with the method comments. Please add comments in the places where you think it is necessary (to explain certain behaviors). First let's do that with Reporting steps and see. However, I do not consider this as a beginner task, since the person have to have a good understanding for the project and the code base to supply meaningful comments. But we can add them as an intermediate task so that many can join and share.

**Comment by sandarumk** [comment]
For the licencing I created #180. We will focus on that with our next meetings with Peace Corps and lead community members.

**Comment by chamika** [comment]
Reporting steps module has the comments and it lacks only class comments according to the template. Hence, I added the comments for the circle of trust module.
See #188

**Comment by sandarumk** [comment]
Okay. I will now tag this issue as free so the new comers can come submit patches by selecting a sub module. The remaining sub modules are 

- [ ] reporting

- [ ] Safety resources

- [ ] Remaining classes

Beginners (who do not have a merged PR) can claim one of this in the comments and submit a PR. When claiming please do not claim a sub module where one have already claimed. To add the methods and comments please follow the template given above and the "Circle of Trust" module. If you have any concerns or questions ask them in the comments or in the slack channel.

Please make one commit per sub module.

**Comment by yogeshbalan** [comment]
I claim reporting sub-module @sandarumk .

**Comment by sandarumk** [comment]
Okay you can proceed. Please refer to Chamika's commit and the template which is described here.

**Comment by minervapanda** [comment]
I claim safety resources submodule @sandarumk

**Comment by sandarumk** [comment]
okay, you can proceed. Please read the previous comments for a better understanding.

**Comment by yogeshbalan** [comment]
@sandarumk , can I send another PR for remaining classes (sub-module).

**Comment by sandarumk** [comment]
Let's just save it for new comers, since you already have one.

**Comment by shamathmika** [comment]
@sandarumk Can I work on remaining classes sub-module?

**Comment by sandarumk** [comment]
Sure. FYI: Remaining classes is not a sub-module. It represents the remaining classes which does not belong to a sub module.

**Comment by TechieToby** [comment]
@sandarumk can the remaining classes be listed, and can i work on one of them?

**Comment by sandarumk** [comment]
There is not much to work for two people in the remaining classes. But you can go through the code and find a sub module which is added recently and missing comments. Then let's see what can be done. If you find more than one sub module with missing comments list them all in your comment. So we can share  the list with other new contributors as well while you are working on one.

**Comment by shamathmika** [comment]
@sandarumk 
Oh, alright! Thank you.

**Comment by shamathmika** [comment]
@sandarumk 
Almost all classes have already been commented. According to the code conventions stated by Oracle, the documentation comments must be
_/******
*Comments
*/_
However documentation comments are made like block comments
_/*
*Comments
*/_
Can I change these?

Also, what about comments describing each method? Aren't they necessary?

**Comment by shamathmika** [comment]
@sandarumk 
Almost all classes have already been commented. According to the code conventions stated by Oracle, the documentation comments must be
_/**
*Comments
*/_
However documentation comments are made like block comments
_/*
*Comments
*/_
Can I change these?

Also, what about comments describing each method? Aren't they necessary?

**Comment by sandarumk** [comment]
@shamathmika In remaining classes there are classes without proper class comments and method comments for example MainActivity.java. You can go through them and follow the format discussed above. You can also go through the previous PRs submitted for this issue in order to better understand what to be done. For example MainActivity.java do not have class comments. So you can add that. It also has a method prepareListData whose method comment is not in the discussed format. You can fix that.

**Comment by shamathmika** [comment]
@sandarumk 
Cool, thanks.

**Comment by top-gun007** [comment]
@sandarumk can I take the policies_glossary submodule

**Comment by sandarumk** [comment]
yeah sure. @chizoba you can do the same with another sub module if any is missing comments.

**Comment by chizoba** [comment]
@sandarumk Okay, will do

**Comment by top-gun007** [comment]
@sandarumk I am done with commenting the policies_glossary sub module should I send the PR?

**Comment by sandarumk** [comment]
yeah sure.

**Comment by chizoba** [comment]
@sandarumk i am done fixing comments for MainActivity.java. I'm currently sending a PR.

**Comment by sandarumk** [comment]
@chizoba main activity belongs to the remaining classes which is assigned to @shamathmika . What I asked to do is choose another sub module with missing comments, claim it here and once approved continue on sending the PR.

**Comment by shamathmika** [comment]
@sandarumk 
I have finished adding method comments to MainActivity. I have also made the documentation comments according to standards in the rest of the modules, and removed unused import statements.
Shall I send a PR? 
Also, I did not know the author and date created for this class, and hence didn't mention them.

**Comment by sandarumk** [comment]
@shamathmika you can submit a PR for the remaining classes. Date and the author can be deducted from the git history. Send the PR after you complete. Remaining sub modules can be assigned to the other contributors who will be willing to contribute.

**Comment by shamathmika** [comment]
Added the required comments, edited comment formats and removed unused import statements #281

**Comment by top-gun007** [comment]
@sandarumk can I work on adding comments to safety_tools sub module and then later send a PR after you have fixed the issues with the repo.

**Comment by sandarumk** [comment]
@top-gun007 Since you already did one regarding the comments, shall we leave it for other new contributors? In the mean time you can select a lint issue and fix it once the issue list is updated.

**Comment by top-gun007** [comment]
ok @sandarumk I will start right away once the list gets updated.
